### ISSUE TITLE
Visual Studio RTM

### ISSUE THREAD

**Issue by dodyg** [frhagn/Typewriter#41]
This extension stops working on the RTM version of Visual Studio 2015. it might be just my installation. Is there any log that I can check?

**Comment by frhagn** [comment]
Hi @dodyg!

Support for Visual Studio 2015 RTM is coming in Typewriter 1.0 which will be released within a couple of weeks. I have published a beta version under Releases here on GitHub. It's almost feature complete and it's pretty stable so you should be able to use it until the final version is released.

There are a couple of changes to the metadata model and custom methods are declared using proper method syntax e.g. `string LoudName(Class c) { return c.Name.ToUpper(); }` instead of the lambda syntax used in the preview. (You can look under closed issues to see more details of what's changed)

Give it a try and please let me know if you have any issues with it.

**Comment by dodyg** [comment]
I receive this error 13:44:30 ERROR: Template error: CS0825 The contextual keyword 'var' may only appear within a local variable declaration or in script code


```
${
    var IsOptional = (Property p) => p.Attributes.Any(a => a.Name == "Optional");
}

declare module R {
$Classes(MigrantFootprints.Core.REST.*)[
    export interface $Name extends $BaseClass[$Name] {
        $Properties[$Name$IsOptional[?]: $Type;
		]
    }
]

$Enums(MigrantFootprints.Core.REST.*)[
    export enum $Name {$Values[
        $Name = $Value,]
    }]
}
```

**Comment by dodyg** [comment]
Ok this works

```
${
    bool IsOptional (Property p)
    { 
        return p.Attributes.Any(a => a.Name == "Optional"); 
    }
}

declare module R {
$Classes(MigrantFootprints.Core.REST.*)[
    export interface $Name extends $BaseClass[$Name] {
        $Properties[$Name$IsOptional[?]: $Type;
		]
    }
]

$Enums(MigrantFootprints.Core.REST.*)[
    export enum $Name {$Values[
        $Name = $Value,]
    }]
}
```

**Comment by dodyg** [comment]
OK I solve the issue by using a custom method as follows

```
${
    bool IsOptional (Property p)
    { 
        return p.Attributes.Any(a => a.Name == "Optional"); 
    }

    string ParentClass(Class c)
    {
        if (c == null || c.BaseClass == null || string.IsNullOrWhiteSpace(c.BaseClass.Name))
            return "Object";

        return c.BaseClass.Name;
    }
}


declare module R {
$Classes(MigrantFootprints.Core.REST.*)[
    export interface $Name extends $ParentClass {
        $Properties[$Name$IsOptional[?]: $Type,
		]
    }
]


$Enums(MigrantFootprints.Core.REST.*)[
    export enum $Name {$Values[
        $Name = $Value,]
    }]
}
```

**Comment by dodyg** [comment]
There's more problem. This template cannot handle enum
```
$Enums(MigrantFootprints.Core.REST.*)[
    export enum $Name {$Values[
        $Name = $Value,]
    }]
}
```
```
public enum TransportationSurfaceType
    {
        Land = 'L',
        Water = 'W',
        Air = 'A'
    }
```

**Comment by frhagn** [comment]
I've published an updated beta release (0.9.21) that resolves the enum bug.
The issue with `$BaseClass` is by design, so if you want to extend `Object` your extension method solution is the way to go.

Please give it a try and let me know if there are any more issues :)

**Comment by dodyg** [comment]
OK the enum issue is fixed. Thank you.
